### ISSUE TITLE
Roads statistics per area

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#3236]
Another quirky proposal (for now just getting your thoughts before I'd flesh it out more systematically):

some data analysis requires something like the output of `/locate` but not point-for-point, rather "show me road properties inside this polygon" including geometry. Instead of the very detailed output of `/locate` I'm rather thinking of a similar output as `/expansion` (extended GeoJSON as per #3206 ), where the `properties` map can be configured in the request with e.g.:
- elevation info: grade, mean_elevation, max_*_slope (that's the requirement where this proposal comes from)
- `max_speed` (esp if I ever finish #2910 )
- `lane_count`
- others?

This could then be another endpoint `roads_stats` or so, taking similar parameters as `/locate` but a polygon instead of locations. 

While this info (esp the one I'd be interested in: elevation related props) doesn't need any routing algos, it'd still be nice to integrate it in a routing engine for the same reasons as `/locate` IMO. Though I realize this is another endpoint possibly blowing up in response size..

**Comment by nilsnolde** [comment]
I started the implementation and realize how many changes this might end up with for a probably dubious outcome utility.. e.g. I think each costing model needs another `Allowed` overload to filter the edges which respect the `costing_options` where applicable (i.e. we couldn't check here for contextual stuff like turn restrictions, uturns etc, since there's no routing).

I'll close for now, but implement it anyways. If someone stumbles upon this and is interested, I'll reference the branch in this issue when I'm done.

**Comment by nilsnolde** [comment]
I thought about using the `locate` service for this, but when I realized it's more changes than I'd like (you wouldn't be happy with that diff;)), no matter how I do it, I figured to just go the least-resistance way of adding a new service, opting for manual labor vs headaches of design/arch decisions;) I'll contain it all in one commit based on the latest master at that time so it's easy to figure out what was done exactly.

haha vector tiles keep coming back! that'd be amazing, wish I'd have the time to meaningfully contribute there. I guess quite some stuff could be borrowed from osrm's tile service..
